src/ripple/rpc/handlers/AccountObjects.cpp

```diff
@@ -72,12 +72,49 @@ Json::Value doAccountObjects (RPC::Context& context)
     if (! ledger->exists(keylet::account (accountID)))
         return rpcError (rpcACT_NOT_FOUND);
 
-    auto [rpcStatus, type] = RPC::chooseLedgerEntryType(params);
-    if (rpcStatus)
+    std::optional<std::vector<LedgerEntryType>> typeFilter;
+
+    if (params.isMember(jss::deletion_blockers_only) &&
+        params[jss::deletion_blockers_only].asBool())
     {
-        result.clear();
-        rpcStatus.inject(result);
-        return result;
+        struct {
+            Json::StaticString name;
+            LedgerEntryType type;
+        } static constexpr deletionBlockers[] = {
+            { jss::check,           ltCHECK },
+            { jss::escrow,          ltESCROW },
+            { jss::payment_channel, ltPAYCHAN },
+            { jss::state,           ltRIPPLE_STATE }
+        };
+
+        typeFilter.emplace();
+        typeFilter->reserve(std::size(deletionBlockers));
+
+        for (auto [name, type] : deletionBlockers)
+        {
+            if (params.isMember(jss::type) &&
+                name != params[jss::type])
```

Comment:
This is reasonable behavior. We could also consider not allowing both `type` and `deletion_blockers_only: true`.